File: bestia-models/src/test/java/net/bestia/model/geometry/SizeTest.java
package net.bestia.model.geometry;

import java.io.Serializable;

import org.junit.Assert;
import org.junit.Test;

public class SizeTest {
	
	@Test
	public void is_serializable() {
		Assert.assertTrue(Serializable.class.isAssignableFrom(Size.class));
	}
	
	@Test(expected=IllegalArgumentException.class)
	public void ctor_negativeValue_throws() {
		new Size(0, -10);
	}
	
	@Test
	public void getter_ctor() {
		Size s = new Size(123, 10);
		Assert.assertEquals(123, s.getWidth());
		Assert.assertEquals(10, s.getHeight());
	}
}

File: bestia-models/src/test/kotlin/net/bestia/model/geometry/SizeTest.kt
package net.bestia.model.geometry

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test
import java.io.Serializable

class SizeTest {

  @Test
  fun is_serializable() {
    assertTrue(Serializable::class.java.isAssignableFrom(Size::class.java))
  }

  @Test
  fun ctor_negativeValue_throws() {
    assertThrows(java.lang.IllegalArgumentException::class.java) {
      Size(0, -10)
    }
  }

  @Test
  fun getter_ctor() {
    val (width, height) = Size(123, 10)
    assertEquals(123, width)
    assertEquals(10, height)
  }

  @Test
  fun equal() {
    val s1 = Size(10, 5)
    val s2 = Size(10, 5)
    val s3 = Size(3, 1)

    assertTrue(s1 == s2)
    assertTrue(s1 == s1)
    assertFalse(s2 == s3)
  }
}
